src/display/annotation_layer.js

```diff
@@ -363,6 +366,50 @@ class AnnotationElement {
     unreachable("Abstract method `AnnotationElement.render` called");
   }
 
+  /**
+   * @private
+   * @returns {Array}
+   */
+  _getElementsByName(name, skipId = null) {
+    const fields = [];
+
+    if (this._fieldObjects) {
```

Comment:
Why both `_fieldObjects` and DOM-based lookup?

Reply:
As mentioned in https://github.com/mozilla/pdf.js/pull/14023#issue-995915957, that's to ensure that the standalone viewer components won't break.
See https://github.com/mozilla/pdf.js/tree/master/examples/components, and in particular the `pageviewer` example, which uses a simplified AnnotationLayer-initialization without all of the parameters that the default viewer will provide.